crates/chainio/clients/elcontracts/src/reader.rs

```diff
@@ -906,7 +905,8 @@ impl ELChainReader {
     ) -> Result<Vec<Address>, ElContractsError> {
         let provider = get_provider(&self.provider);
 
-        let contract_allocation_manager = AllocationManager::new(self.allocation_manager, provider);
+        let contract_allocation_manager =
+            AllocationManager::new(self.allocation_manager.unwrap(), provider);
```

Comment:
This should map into an error with `ok_or` and return with `?` instead of using `unwrap`.